src/Pos/DB/Block.hs

```diff
@@ -221,14 +224,21 @@ putBi k v = rocksPutBi k v =<< getBlockIndexDB
 delete :: (MonadDB m) => ByteString -> m ()
 delete k = rocksDelete k =<< getBlockIndexDB
 
-getData ::  (MonadIO m, Bi v) => FilePath -> m (Maybe v)
-getData = undefined
+getData ::  (MonadIO m, MonadCatch m, Bi v) => FilePath -> m (Maybe v)
+getData fp = liftIO (decodeMaybe . BSL.fromStrict <$> BS.readFile fp) `catch` handle
+  where
+    decodeMaybe x = case decodeFull x of
+        Left _  -> Nothing
+        Right r -> Just r
+    handle (_::SomeException) = pure Nothing
```

Comment:
I don't like it to be honest. We can't distinguish situations when there is no data/when it's malformed (can't be parsed)/when hard drive is broken. I think it's bad approach.